packages/js/product-editor/src/components/header/hooks/use-publish/use-publish.tsx

```diff
@@ -96,11 +102,25 @@ export function usePublish( {
 							? 'product_publish_error'
 							: 'product_create_error',
 					} as WPError;
-					if ( ( error as Record< string, string > ).variations ) {
+					let isPriceError: string | undefined;
+					Object.entries( error as Record< string, string > ).forEach(
```

Comment:
We should not do the error handling this way since this does not scale for other validations messages different than regular_price. What about the other fields that trigger validation messages too. This must consider all the possible fields with error not only the regular_price.

Reply:
Yeah, I thought that same thing. Since we already had an exception for variable products I thought, we could continue on that path. But I prefer to show the message for the other fields as well.
Let's confirm with @jarekmorawski and @AnnaMag

Reply:
@mdperez86 @octaedro sounds reasonable.

Reply:
Imho, error handling should be uniform across the whole form to lay the foundation for https://github.com/woocommerce/woocommerce/issues/40371 when we're ready. In this case, we should treat the notice above the table as a validation message.

Just like elsewhere, we take people to the affected tab/field, when someone attempts to publish a product with variations without prices, we should take them to the Variations tab and scroll to the Variations table, so the notice is clearly visible. I don't think we need the snackbar unless it's required for accessibility.

Later on, the message we show in the message will also be displayed in the https://github.com/woocommerce/woocommerce/issues/40371.